examples/pure-module/README.md

```diff
@@ -0,0 +1,282 @@
+This example features the `pure-module` flag.
+
+The example contains a assumingly big library named `big-module`. This library is split into multiple child modules `a`, `b` and `c`. The exports from these modules are reexported in the entry (`index.js`) of the library. A consumer uses **some** of the exports by importing them from the library via `import { a, b } from "big-module"`. In this case according to the EcmaScript spec all child modules must be evaluated because they could contain side effects.
+
+When using the `"pure-module": true` flag in `package.json` the package author promises that modules contain no side effects expect exposed exports. This allows to optimize reexports. In this case `import { a, b } from "big-module-pure"` is treated like `import { a } from "big-module-pure/a"; import { b } from "big-module-pure/b"`. This is done by following reexports.
```

Comment:
no side effects ~~expect~~ *except* exposed exports
This allows *it* to optimize reexports
And I think you mean: `import { a, b } from "big-module-pure"` is treated like `import a from "big-module-pure/a"; import b from "big-module-pure/b"`